test/pkg.jl

```diff
@@ -139,6 +139,11 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+    Pkg.checkout("Example", "test-branch") #17364
```

Comment:
or commit anything to it? maybe we could construct a local branch artificially, or does does the bug require it to be a remote tracking thing?